src/main/java/org/folio/entlinks/service/authority/AuthoritySourceFileService.java

```diff
@@ -184,6 +184,16 @@ public boolean authoritiesExistForSourceFile(UUID sourceFileId) {
     return authorityRepository.existsAuthorityByAuthoritySourceFileId(sourceFileId);
   }
 
+  public boolean authoritiesExistForSourceFile(UUID sourceFileId, String tenantId) {
+    if (sourceFileId == null || tenantId == null) {
+      return false;
+    }
+
+    var command = String.format("select exists (select true from %s.authority a where a.source_file_id='%s')",
```

Comment:
```suggestion
    var command = String.format("select exists (select true from %s.authority a where a.source_file_id='%s' limit 1)",
```
I think it will be more efficient because we don't need all of the auhtorities, at least one.